### ISSUE TITLE
[SPIKE] Investigate how to get codecoverage for n_components and check `build_conda_pkg` tests

### ISSUE THREAD

**Issue by ParthivNaresh** [alteryx/evalml#2555]
This issue covers a couple of points:

1. Due to the various tests we have for the components we leverage in different environments and the codecoverage on tests we perform, we end up not covering `is_using_conda` and `is_running_py_39_or_above`. Therefore they fail to ensure complete code coverage when we edit these values.
```
def test_all_components(
    has_minimal_dependencies, is_running_py_39_or_above, is_using_conda
):
    if has_minimal_dependencies:
        n_components = 37
    elif is_using_conda:
        n_components = 48
    elif is_running_py_39_or_above:
        n_components = 47
    else:
        n_components = 49
    assert len(all_components()) == n_components
```

2. Ensure that merging this to main doesn't break the `build_conda_package` process, especially regarding the [passing](https://github.com/conda-forge/evalml-core-feedstock/blob/latest_release_changes/recipe/meta.yaml#L57) of `--has-minimal-dependencies` to `evalml-core` and `--is-using-conda` to `evalml`

**Comment by dsherry** [comment]
Yep. First step is to define why we need this test, or whether we should delete it and write some other coverage for `all_components()`
